### ISSUE TITLE
Errors while reading frames for Zen Thermostat

### ISSUE THREAD

**Issue by mrstegeman** [mozilla-iot/zigbee-adapter#143]
```
2019-07-01 19:55:26.316 INFO   : zigbee: ZigbeeNode created: addr64: 002446000011707b addr16: 3cfc
2019-07-01 19:55:29.575 INFO   : zigbee: zb-002446000011707b-thermostat property: temperature profileId: 0104 endpoint: 1 clusterId: 0201 readRsp value: 23.1
2019-07-01 19:55:29.718 INFO   : zigbee: zb-002446000011707b-thermostat property: mode profileId: 0104 endpoint: 1 clusterId: 0201 readRsp value: Heat (4)
2019-07-01 19:55:32.880 ERROR  : zigbee: Error parsing ZCL frame
2019-07-01 19:55:32.894 ERROR  : zigbee: Error: parse timeout
2019-07-01 19:55:32.896 ERROR  : zigbee:     at Dissolve.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:234:22)
2019-07-01 19:55:32.897 ERROR  : zigbee:     at Object.onceWrapper (events.js:315:30)
2019-07-01 19:55:32.899 ERROR  : zigbee:     at emitOne (events.js:116:13)
2019-07-01 19:55:32.901 ERROR  : zigbee:     at Dissolve.emit (events.js:211:7)
2019-07-01 19:55:32.902 ERROR  : zigbee:     at Timeout.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:220:20)
2019-07-01 19:55:32.904 ERROR  : zigbee:     at ontimeout (timers.js:498:11)
2019-07-01 19:55:32.906 ERROR  : zigbee:     at tryOnTimeout (timers.js:323:5)
2019-07-01 19:55:32.907 ERROR  : zigbee:     at Timer.listOnTimeout (timers.js:290:5)
2019-07-01 19:55:32.909 ERROR  : zigbee: { type: 145,
2019-07-01 19:55:32.911 ERROR  : zigbee:   remote64: '002446000011707b',
2019-07-01 19:55:32.913 ERROR  : zigbee:   remote16: '3cfc',
2019-07-01 19:55:32.914 ERROR  : zigbee:   sourceEndpoint: '01',
2019-07-01 19:55:32.917 ERROR  : zigbee:   destinationEndpoint: '01',
2019-07-01 19:55:32.919 ERROR  : zigbee:   clusterId: '0201',
2019-07-01 19:55:32.931 ERROR  : zigbee:   profileId: '0104',
2019-07-01 19:55:32.937 ERROR  : zigbee:   receiveOptions: 65,
2019-07-01 19:55:32.939 ERROR  : zigbee:   data: <Buffer 18 07 07 00 00 1e 00> }
2019-07-01 19:55:42.882 ERROR  : zigbee: Error parsing ZCL frame
2019-07-01 19:55:42.886 ERROR  : zigbee: Error: parse timeout
2019-07-01 19:55:42.888 ERROR  : zigbee:     at Dissolve.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:234:22)
2019-07-01 19:55:42.889 ERROR  : zigbee:     at Object.onceWrapper (events.js:315:30)
2019-07-01 19:55:42.891 ERROR  : zigbee:     at emitOne (events.js:116:13)
2019-07-01 19:55:42.893 ERROR  : zigbee:     at Dissolve.emit (events.js:211:7)
2019-07-01 19:55:42.898 ERROR  : zigbee:     at Timeout.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:220:20)
2019-07-01 19:55:42.899 ERROR  : zigbee:     at ontimeout (timers.js:498:11)
2019-07-01 19:55:42.901 ERROR  : zigbee:     at tryOnTimeout (timers.js:323:5)
2019-07-01 19:55:42.902 ERROR  : zigbee:     at Timer.listOnTimeout (timers.js:290:5)
2019-07-01 19:55:42.904 ERROR  : zigbee: { type: 145,
2019-07-01 19:55:42.906 ERROR  : zigbee:   remote64: '002446000011707b',
2019-07-01 19:55:42.907 ERROR  : zigbee:   remote16: '3cfc',
2019-07-01 19:55:42.909 ERROR  : zigbee:   sourceEndpoint: '01',
2019-07-01 19:55:42.910 ERROR  : zigbee:   destinationEndpoint: '01',
2019-07-01 19:55:42.911 ERROR  : zigbee:   clusterId: '0201',
2019-07-01 19:55:42.913 ERROR  : zigbee:   profileId: '0104',
2019-07-01 19:55:42.914 ERROR  : zigbee:   receiveOptions: 65,
2019-07-01 19:55:42.916 ERROR  : zigbee:   data: <Buffer 18 07 07 00 00 1e 00> }
2019-07-01 19:55:53.161 ERROR  : zigbee: Error parsing ZCL frame
2019-07-01 19:55:53.164 ERROR  : zigbee: Error: parse timeout
2019-07-01 19:55:53.165 ERROR  : zigbee:     at Dissolve.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:234:22)
2019-07-01 19:55:53.167 ERROR  : zigbee:     at Object.onceWrapper (events.js:315:30)
2019-07-01 19:55:53.168 ERROR  : zigbee:     at emitOne (events.js:116:13)
2019-07-01 19:55:53.170 ERROR  : zigbee:     at Dissolve.emit (events.js:211:7)
2019-07-01 19:55:53.171 ERROR  : zigbee:     at Timeout.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:220:20)
2019-07-01 19:55:53.173 ERROR  : zigbee:     at ontimeout (timers.js:498:11)
2019-07-01 19:55:53.174 ERROR  : zigbee:     at tryOnTimeout (timers.js:323:5)
2019-07-01 19:55:53.176 ERROR  : zigbee:     at Timer.listOnTimeout (timers.js:290:5)
2019-07-01 19:55:53.177 ERROR  : zigbee: { type: 145,
2019-07-01 19:55:53.178 ERROR  : zigbee:   remote64: '002446000011707b',
2019-07-01 19:55:53.180 ERROR  : zigbee:   remote16: '3cfc',
2019-07-01 19:55:53.181 ERROR  : zigbee:   sourceEndpoint: '01',
2019-07-01 19:55:53.183 ERROR  : zigbee:   destinationEndpoint: '01',
2019-07-01 19:55:53.184 ERROR  : zigbee:   clusterId: '0201',
2019-07-01 19:55:53.186 ERROR  : zigbee:   profileId: '0104',
2019-07-01 19:55:53.187 ERROR  : zigbee:   receiveOptions: 65,
2019-07-01 19:55:53.188 ERROR  : zigbee:   data: <Buffer 18 07 07 00 00 1e 00> }
2019-07-01 19:56:43.565 INFO   : zigbee: handleQueryNextImageReq: rspFrame =
2019-07-01 19:56:43.568 INFO   : zigbee: { type: 145,
2019-07-01 19:56:43.570 INFO   : zigbee:   remote64: '002446000011707b',
2019-07-01 19:56:43.571 INFO   : zigbee:   remote16: '3cfc',
2019-07-01 19:56:43.573 INFO   : zigbee:   sourceEndpoint: '01',
2019-07-01 19:56:43.574 INFO   : zigbee:   destinationEndpoint: '01',
2019-07-01 19:56:43.576 INFO   : zigbee:   clusterId: '0019',
2019-07-01 19:56:43.577 INFO   : zigbee:   profileId: '0104',
2019-07-01 19:56:43.579 INFO   : zigbee:   receiveOptions: 65,
2019-07-01 19:56:43.580 INFO   : zigbee:   data: <Buffer 11 01 01 01 58 11 03 00 10 02 00 00 03 00>,
2019-07-01 19:56:43.582 INFO   : zigbee:   zcl:
2019-07-01 19:56:43.583 INFO   : zigbee:    { frameCntl: { frameType: 1, manufSpec: 0, direction: 0, disDefaultRsp: 1 },
2019-07-01 19:56:43.585 INFO   : zigbee:      manufCode: 0,
2019-07-01 19:56:43.586 INFO   : zigbee:      seqNum: 1,
2019-07-01 19:56:43.588 INFO   : zigbee:      cmdId: 'queryNextImageReq',
2019-07-01 19:56:43.589 INFO   : zigbee:      payload: { fieldCtrl: 1, manufCode: 4440, imageType: 3 } },
2019-07-01 19:56:43.591 INFO   : zigbee:   zclSeqNum: 1,
2019-07-01 19:56:43.592 INFO   : zigbee:   zclCmdId: 'queryNextImageReq' }
2019-07-01 19:56:47.066 ERROR  : zigbee: Error parsing ZCL frame
2019-07-01 19:56:47.069 ERROR  : zigbee: Error: parse timeout
2019-07-01 19:56:47.070 ERROR  : zigbee:     at Dissolve.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:234:22)
2019-07-01 19:56:47.072 ERROR  : zigbee:     at Object.onceWrapper (events.js:315:30)
2019-07-01 19:56:47.073 ERROR  : zigbee:     at emitOne (events.js:116:13)
2019-07-01 19:56:47.075 ERROR  : zigbee:     at Dissolve.emit (events.js:211:7)
2019-07-01 19:56:47.076 ERROR  : zigbee:     at Timeout.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:220:20)
2019-07-01 19:56:47.078 ERROR  : zigbee:     at ontimeout (timers.js:498:11)
2019-07-01 19:56:47.079 ERROR  : zigbee:     at tryOnTimeout (timers.js:323:5)
2019-07-01 19:56:47.080 ERROR  : zigbee:     at Timer.listOnTimeout (timers.js:290:5)
2019-07-01 19:56:47.082 ERROR  : zigbee: { type: 145,
2019-07-01 19:56:47.083 ERROR  : zigbee:   remote64: '002446000011707b',
2019-07-01 19:56:47.085 ERROR  : zigbee:   remote16: '3cfc',
2019-07-01 19:56:47.086 ERROR  : zigbee:   sourceEndpoint: '01',
2019-07-01 19:56:47.088 ERROR  : zigbee:   destinationEndpoint: '01',
2019-07-01 19:56:47.089 ERROR  : zigbee:   clusterId: '0201',
2019-07-01 19:56:47.090 ERROR  : zigbee:   profileId: '0104',
2019-07-01 19:56:47.092 ERROR  : zigbee:   receiveOptions: 65,
2019-07-01 19:56:47.093 ERROR  : zigbee:   data: <Buffer 18 08 07 00 00 1e 00> }
2019-07-01 19:56:56.848 ERROR  : zigbee: Error parsing ZCL frame
2019-07-01 19:56:56.851 ERROR  : zigbee: Error: parse timeout
2019-07-01 19:56:56.852 ERROR  : zigbee:     at Dissolve.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:234:22)
2019-07-01 19:56:56.853 ERROR  : zigbee:     at Object.onceWrapper (events.js:315:30)
2019-07-01 19:56:56.855 ERROR  : zigbee:     at emitOne (events.js:116:13)
2019-07-01 19:56:56.857 ERROR  : zigbee:     at Dissolve.emit (events.js:211:7)
2019-07-01 19:56:56.858 ERROR  : zigbee:     at Timeout.<anonymous> (/home/pi/.mozilla-iot/addons/zigbee-adapter/node_modules/zcl-packet/lib/foundation.js:220:20)
2019-07-01 19:56:56.859 ERROR  : zigbee:     at ontimeout (timers.js:498:11)
2019-07-01 19:56:56.861 ERROR  : zigbee:     at tryOnTimeout (timers.js:323:5)
2019-07-01 19:56:56.862 ERROR  : zigbee:     at Timer.listOnTimeout (timers.js:290:5)
2019-07-01 19:56:56.864 ERROR  : zigbee: { type: 145,
2019-07-01 19:56:56.865 ERROR  : zigbee:   remote64: '002446000011707b',
2019-07-01 19:56:56.867 ERROR  : zigbee:   remote16: '3cfc',
2019-07-01 19:56:56.868 ERROR  : zigbee:   sourceEndpoint: '01',
2019-07-01 19:56:56.870 ERROR  : zigbee:   destinationEndpoint: '01',
2019-07-01 19:56:56.871 ERROR  : zigbee:   clusterId: '0201',
2019-07-01 19:56:56.872 ERROR  : zigbee:   profileId: '0104',
2019-07-01 19:56:56.874 ERROR  : zigbee:   receiveOptions: 65,
2019-07-01 19:56:56.875 ERROR  : zigbee:   data: <Buffer 18 08 07 00 00 1e 00> }
2019-07-01 19:57:10.354 INFO   : zigbee: zb-002446000011707b-thermostat property: temperature profileId: 0104 endpoint: 1 clusterId: 0201 report value: 23
2019-07-01 19:57:40.108 INFO   : zigbee: Unable to find node for remote16 = fffd
```

**Comment by mrstegeman** [comment]
`zb-0013a20040a3b2af.json` contains the following:

```json
{
  "driver": {
    "deviceType": "0x3000c",
    "deviceTypeStr": "XStick",
    "serialNumber": "0013a20040a3b2af",
    "nodeIdentifier": "",
    "configuredPanId64": "85bcc9ccbcbae067",
    "operatingPanId64": "85bcc9ccbcbae067",
    "operatingPanId16": "1c19",
    "operatingChannel": 15,
    "scanChannels": "0x1ffe",
    "apiOptions": 3
  },
  "nodes": {
    "002446000011707b": {
      "addr64": "002446000011707b",
      "addr16": "3cfc",
      "activeEndpoints": {
        "1": {
          "profileId": "0104",
          "deviceId": "0301",
          "deviceVersion": 0,
          "inputClusters": [
            "0000",
            "0001",
            "0003",
            "0004",
            "0005",
            "0020",
            "0201",
            "0202",
            "0204",
            "0b05"
          ],
          "outputClusters": [
            "000a",
            "0019"
          ],
          "classifierAttributesPopulated": true
        }
      },
      "properties": {
        "temperature": {
          "name": "temperature",
          "value": 23,
          "visible": true,
          "title": "Temperature",
          "type": "number",
          "@type": "TemperatureProperty",
          "unit": "degree celsius",
          "minimum": 0,
          "maximum": 40,
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "localTemp",
          "attrId": 0,
          "fireAndForget": false,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "mode": {
          "name": "mode",
          "value": "Heat",
          "visible": true,
          "title": "Mode",
          "type": "string",
          "enum": [
            "Off",
            "Auto",
            "Cool",
            "Heat"
          ],
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "systemMode",
          "attrId": 28,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "runMode": {
          "name": "runMode",
          "visible": true,
          "title": "Run Mode",
          "type": "string",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "runningMode",
          "attrId": 30,
          "fireAndForget": false,
          "bindNeeded": false,
          "configReportNeeded": true,
          "initialReadNeeded": true
        },
        "_deadband": {
          "name": "_deadband",
          "visible": false,
          "title": "DeadBand",
          "type": "number",
          "unit": "degree celsius",
          "minimum": 1,
          "maximum": 2.5,
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "minSetpointDeadBand",
          "attrId": 25,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_absMaxHeatTarget": {
          "name": "_absMaxHeatTarget",
          "visible": false,
          "title": "Abs Max Heat Target",
          "type": "number",
          "unit": "degree celsius",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "absMaxHeatSetpointLimit",
          "attrId": 4,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_absMinHeatTarget": {
          "name": "_absMinHeatTarget",
          "visible": false,
          "title": "Abs Min Heat Target",
          "type": "number",
          "unit": "degree celsius",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "absMinHeatSetpointLimit",
          "attrId": 3,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_maxHeatTarget": {
          "name": "_maxHeatTarget",
          "visible": false,
          "title": "Max Heat Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "maxHeatSetpointLimit",
          "attrId": 22,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_minHeatTarget": {
          "name": "_minHeatTarget",
          "visible": false,
          "title": "Min Heat Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "minHeatSetpointLimit",
          "attrId": 21,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "heatTarget": {
          "name": "heatTarget",
          "visible": true,
          "title": "Heat Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "occupiedHeatingSetpoint",
          "attrId": 18,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_absMaxCoolTarget": {
          "name": "_absMaxCoolTarget",
          "visible": false,
          "title": "Abs Max Cool Target",
          "type": "number",
          "unit": "degree celsius",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "absMaxCoolSetpointLimit",
          "attrId": 6,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_absMinCoolTarget": {
          "name": "_absMinCoolTarget",
          "visible": false,
          "title": "Abs Min Cool Target",
          "type": "number",
          "unit": "degree celsius",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "absMinCoolSetpointLimit",
          "attrId": 5,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_maxCoolTarget": {
          "name": "_maxCoolTarget",
          "visible": false,
          "title": "Max Cool Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "maxCoolSetpointLimit",
          "attrId": 24,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_minCoolTarget": {
          "name": "_minCoolTarget",
          "visible": false,
          "title": "Min Cool Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "minCoolSetpointLimit",
          "attrId": 23,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "coolTarget": {
          "name": "coolTarget",
          "visible": true,
          "title": "Cool Target",
          "type": "number",
          "unit": "degree celsius",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "occupiedCoolingSetpoint",
          "attrId": 17,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "hold": {
          "name": "hold",
          "visible": true,
          "title": "Hold",
          "type": "boolean",
          "@type": "BooleanProperty",
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 513,
          "attr": "tempSetpointHold",
          "attrId": 35,
          "fireAndForget": false,
          "bindNeeded": false,
          "configReportNeeded": true,
          "initialReadNeeded": true
        },
        "units": {
          "name": "units",
          "visible": true,
          "title": "Units",
          "type": "string",
          "enum": [
            "C",
            "F"
          ],
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 516,
          "attr": "tempDisplayMode",
          "attrId": 0,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "_fanModeSeq": {
          "name": "_fanModeSeq",
          "visible": false,
          "title": "Fan Sequence",
          "type": "string",
          "enum": [
            "Low/Medium/High",
            "Low/High",
            "Low/Medium/High/Auto",
            "Low/High/Auto",
            "On/Auto"
          ],
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 514,
          "attr": "fanModeSequence",
          "attrId": 1,
          "fireAndForget": true,
          "bindNeeded": false,
          "configReportNeeded": false,
          "initialReadNeeded": true
        },
        "fanMode": {
          "name": "fanMode",
          "visible": true,
          "title": "Fan",
          "type": "string",
          "enum": [],
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 514,
          "attr": "fanMode",
          "attrId": 0,
          "fireAndForget": false,
          "bindNeeded": true,
          "configReportNeeded": true,
          "initialReadNeeded": true
        },
        "voltage": {
          "name": "voltage",
          "visible": true,
          "title": "Voltage",
          "type": "number",
          "@type": "VoltageProperty",
          "unit": "volt",
          "readOnly": true,
          "profileId": 260,
          "endpoint": 1,
          "clusterId": 1,
          "attr": "batteryVoltage",
          "attrId": 32,
          "fireAndForget": false,
          "bindNeeded": true,
          "configReportNeeded": true,
          "initialReadNeeded": true
        }
      },
      "name": "zb-002446000011707b-thermostat",
      "type": "thing",
      "@type": [
        "TemperatureSensor"
      ],
      "defaultName": "zb-002446000011707b-thing",
      "extendedTimeout": true,
      "activeEndpointsPopulated": true,
      "nodeInfoEndpointsPopulated": true,
      "modelId": "Zen-01",
      "powerSource": 129,
      "checkinInterval": 80,
      "longPollInterval": 56,
      "shortPollInterval": 2,
      "fastPollTimeout": 40,
      "pollCtrlBindingNeeded": false
    }
  }
}
```

**Comment by dhylands** [comment]
It would be good to enable the following debug: frames, frameDetail, rawFrames  while getting the parsing errors.

**Comment by mrstegeman** [comment]
Emailed.

**Comment by dhylands** [comment]
It looks like technically, this is in fact an invalid response.

I'll see if I can put in a workaround for this.

**Comment by mrstegeman** [comment]
Fixed by #145
